packages/grafana-runtime/src/services/dataSourceSrv.ts

```diff
@@ -29,6 +29,11 @@ export interface DataSourceSrv {
    * Get all data sources except for internal ones that usually should not be listed like mixed data source.
    */
   getExternal(): DataSourceInstanceSettings[];
+
+  /**
+   * Get settings by name or uid
+   */
+  getSettingsFor(nameOrUid: string | null | undefined): DataSourceInstanceSettings | undefined;
```

Comment:
I'm a bit confused about the api for `DataSourceSrv `:

Should we deprecate getDataSourceSettingsByUid or have 2 distinct functions? One for uid and one for name? I understand that name is going away, but I think having both uid and name in the same argument is confusing for consumers.

Reply:
I have another PR that touches some stuff related to this function (for a feature it was created for). Was hoping to remove it but might have to deprecate it for a version first
https://github.com/grafana/grafana/pull/29447